### PR TITLE
DBAAS-338

### PR DESCRIPTION
**Pull request by davidwbt**
Additional changes to ensure um2 is setup correctly for replication of subsequent changes.
This is done by adding allowing um1 to connect to um2 as root mariadb user and then add
a shell script which waits for the replication slave status to show up good. All this fun is needed because replication gets setup asynchronously after the system shows active.

### CODE REVIEW

File: columnstore/wait_for_columnstore_active [hunk 1]
```diff
@@ -39,14 +39,19 @@ fi
 if [ -f "/usr/local/mariadb/columnstore/mysql/bin/mysql" ]; then
     echo "Waiting for system catalog to be fully created"
     ATTEMPT=1
-    STATUS=$(/usr/local/mariadb/columnstore/mysql/bin/mysql -u root test -e "drop table if exists installtest; create table installtest(i tinyint) engine=columnstore;")
+    STATUS=$(/usr/local/mariadb/columnstore/mysql/bin/mysql -u root test -e "drop table if exists installtest; create table installtest(i tinyint) engine=columnstore;" 2>&1)
     while [ 1 -eq $? ] && [ $ATTEMPT -le $MAX_TRIES ]; do
         if [ ! -z $CS_DEBUG ]; then
             echo "wait_for_columnstore_active($ATTEMPT/$MAX_TRIES): create table test error: $STATUS"
+        f
```
Comment (izlatoustov) [inline columnstore/wait_for_columnstore_active]: Syntax error it should be **fi**

**Comment by drrtuy** [comment]
It was supposed to facilitate debugging since I could look at the commands that was executed.

**Comment by davidwbt** [comment]
Hi guys good feedback. I'm not sure how this worked for me yesterday. I have fixed the fi typo and also the corresponding if statement.  Also made the dbinit custom sh script logic interpret the CS_DEBUG env var so that it is not noisy by default.  IF you want that you can add CS_DEBUG=1 to the env section of the container in the compose file.   @drrtuy how much memory does your docker env have?  On mine i am using 8gb vm and it normally starts in about 1.5 minute tops hence why i picked 3 mins but we could make the timeout bigger (you can also similarly set the env var CS_WAIT_ATTEMPTS to override the default number.

On the replication, the postConfigure script does this using sh scripts in the cs bin directory. The new wait for script is an attempt to synchronize on that. The um2 script should get run automatically assuming it detects the systems has started. If it doesn't run reliably we should look into this. 

It's an asynchronous mess trying to get this to work :)

**Comment by drrtuy** [comment]
It is 16gb MBP. I presume there will be complains if the TO isn't big enough so there is no reason to increase it now. 
The UM2 must complain about non-existent user applying grants. You could use GRANT..IDENTIFY BY to create/grant in one shot. You won't use this since the root user is passwordless. I bet you need create user root@10.5.0.2 right before the grant statement. Message me if it unnecessary so I could merge the PR.

**Comment by davidwbt** [comment]
grant all on *.* to root@10... is valid short hand to grant and create the user if it does not exist. Are you still seeing issues with this?  The reason this is needed is to allow um1 to connect to um2 as columnstore doesn't set this up by default. It will setup um2 to um1 for replication.  By having this we can essentially have the wait check to make sure replication is enabled before seeding the data. The CS replication scripts currently query the current master binlog pos and start the slave from that. So if the table creation and seeding has already started then you miss that in the replication log.

Although your mbp might have 16gb, by default docker creates i think a 2gb vm - you can check in your docker settings, that might be too low for this.

**Comment by drrtuy** [comment]
I rechecked the grant statement and it work for me also so I proceed with
merging.
Docker uses cgroups to limit resources under the hood and there is no
limitation on memory by default for a container.

**Pull request merged**